File: MyDailyBeat/API/VerveAPI/Stack.m
//
//  Stack.m
//  VerveAPI
//
//  Created by Virinchi Balabhadrapatruni on 6/1/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//

#import "Stack.h"

@implementation Stack

@synthesize top = _top;
@synthesize stack = _stack;

- (id) init {
    self = [super init];
    if (self) {
        self.top = 0;
        self.stack = [[NSMutableArray alloc] init];
    }
    return self;
}

- (void) push: (id) object {
    [self.stack addObject:object];
    self.top = self.top + 1;
}
- (id) pop {
    id retval = [self.stack objectAtIndex:self.top];
    [self.stack removeObjectAtIndex:self.top];
    self.top = self.top - 1;
    return retval;
}

@end

File: MDB_Swift/MyDailyBeat/API/src/Stack.swift
//  Converted with Swiftify v1.0.6242 - https://objectivec2swift.com/
//
//  Stack.swift
//  VerveAPI
//
//  Created by Virinchi Balabhadrapatruni on 6/1/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//
import Foundation
class Stack: NSObject {

    var top: Int = 0
    var stack = [Any]()

    override init() {
        super.init()
        
        self.top = 0
        self.stack = [Any]()
    
    }

    func push(_ object: Any) {
        self.stack.append(object)
        self.top = self.top + 1
    }

    func pop() -> Any {
        let retval: Any? = self.stack[self.top]
        self.stack.remove(at: self.top)
        self.top = self.top - 1
        return retval!
    }
}
